### ISSUE TITLE
Using Zig CC to cross-compile pure C code to an aarch64-macos dylib from x86_64 linux host fails with error: UnexpectedRemainder

### ISSUE THREAD

**Issue by deforde** [ziglang/zig#11907]
### Zig Version

0.10.0-dev.2624+d506275a0

### Steps to Reproduce

On an x86_64 linux host (in my case, specifically: Ubuntu 22.04).

Create a file: `foo.c` with the following content:
```
#include <stdint.h>

uint8_t j = 1;

void foo(int *b)
{
    if(1.0f / j > 1.0f) {
        *b = 1;
    }
}
```

Attempt to cross-compile this source file into an aarch64-macos dylib as follows:
```
zig cc -target aarch64-macos -shared -fPIC foo.c -o libfoo.dylib
```
This should fail with an error:
```
error: UnexpectedRemainder
```

See the following repo: https://github.com/deforde/zig-macho-bug, for a more detailed example, including a Dockerfile to build an image that sets up an environment that can be used to reproduce the issue (and to compare to an equivalent cross-compilation using clang as well as other native builds).

Other details:

Zig:
- zig-linux-x86\_64-0.10.0-dev.2624+d506275a0

Host:
- x86\_64 Ubuntu 22.04

Clang:
- Ubuntu clang version 14.0.0-1ubuntu1
- Target: x86\_64-pc-linux-gnu
- Thread model: posix

MacOS SDK:
- MacOSX11.3
- Available here: https://github.com/phracker/MacOSX-SDKs/releases/download/11.3/MacOSX11.3.sdk.tar.xz

### Expected Behavior

I expected the dylib to be compiled successfully, as `zig` is able to do for a native target (`clang` is also able to both natively and cross-compile the same code successfully).

### Actual Behavior

The compilation failed with the following error:
```
error: UnexpectedRemainder
```

**Comment by kubkon** [comment]
Can you confirm that this is still the case with latest Zig master please?

**Comment by deforde** [comment]
I updated the bug reproduction code located here: https://github.com/deforde/zig-macho-bug, to use zig-linux-x86_64-0.11.0-dev.2375+771d07268, and it is no longer capable of reproducing the issue.

I also went back to the more complex use-case (i.e. building a real, larger and more complex application) for which I originally stumbled upon this issue, and found that the latest build of zig also seems to have resolved the problem there.

So as far as I am concerned, this appears to be resolved.

Thanks!
